react/README.md

```diff
@@ -34,12 +34,28 @@ For example, the following works:
 <MaskedTextInput
   mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]}
   className="form-control"
-  placeholder="Enter zip code"
+  placeholder="Enter a phone number"
   guide={false}
   id="my-input-id"
 />
 ```
 
+You can additionally use your own custom input component by passing in the prop `inputElement`.
```

Comment:
Trying to avoid `findDOMNode` all together on this.
Examples in this thread https://github.com/yannickcr/eslint-plugin-react/issues/678#issue-165177220

The idea would be if a custom input element was passed in we would pass in `inputRef` as a prop that is a callback.

```
<CustomInput
inputRef={(domInput) => this.inputElement = domInput
/>
```

Then it is out of our hands. It is up to the user (since they are taking control) to get the `inputRef` prop that we passed in down to the true DOM element `<input ref={this.props.inputRef} />`

This is what Dan Abramov recommends. It does make it harder to integrate with additional 3rd party libraries. This would ultimately required the use of `cloneElement` which not mayn know about but is extremely handy. Even then would the 3rd party lib to eventually get the `ref` onto the actual DOM node.